communication/nats/discovery/address.go

```diff
@@ -35,10 +37,22 @@ func NewAddress(topic string, addresses ...string) *AddressNATS {
 }
 
 // NewAddressGenerate generates NATS address for current node
-func NewAddressGenerate(brokerIP string, myID identity.Identity) *AddressNATS {
-	address := fmt.Sprintf("nats://%s:%d", brokerIP, BrokerPort)
+func NewAddressGenerate(rawurl string, myID identity.Identity) *AddressNATS {
+	// url.Parse() functions incorrectly if rawurl doesn't start with protocol.
+	if !strings.HasPrefix(rawurl, "nats:") {
+		rawurl = fmt.Sprintf("nats://%s", rawurl)
+	}
+
+	url, err := url.Parse(rawurl)
+	if err != nil {
+		return nil
```

Comment:
We should not just hide an error. 
I'd change the function to return the error, and stop the application if the provided address was incorrect.

Reply:
I actually did that at first.  Then I thought that perhaps it was the callers responsibility to pass in a correct string.  Caller can still tell we failed by checking if returned string is "".  This imitates the way at least one function in the standard library works (I can't think which one off the top of my head though).  Doing it this way will mean we could verify the option strings when the `myst` command first starts.  This makes the rest of the code more simple since we _know_ we have valid option strings.  What do you think?

Reply:
Function naming. GenerateNewAddress ?

Reply:
I'll agree with Dmitry - it's best to return the error here. Having to check for an empty string does not feel like a Go way of doing things. I'd stick with the error being returned and the good old if err != nil.

Reply:
> Function naming. GenerateNewAddress ?
What's about just `NewAddress`?

Reply:
Looks like we have consensus on this then.  Will re-do with an error return.  Thanks

Reply:
@tcharding replying to your question - yes I think GenerateNewAddress is a better name.

Reply:
lol, my dislexia was playing up @tadovas I read your original comment a bunch of times before and didn't realize you had switched the word `Generate` from the back of the function name to the front.  Thanks for clarifying.